### PR TITLE
fix: vvm version detection

### LINKED ISSUE #342: Cannot load contract
```bash
>>> import boa
>>> boa.load("favorites.vy")
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    boa.load("favorites.vy")
    ~~~~~~~~^^^^^^^^^^^^^^^^
  File "/my/path/scrap/.venv/lib/python3.13/site-packages/boa/interpret.py", line 177, in load
    return loads(f.read(), *args, name=name, **kwargs, filename=filename)
  File "/my/path/scrap/.venv/lib/python3.13/site-packages/boa/interpret.py", line 189, in loads
    d = loads_partial(source_code, name, filename=filename, compiler_args=compiler_args)
  File "/my/path/scrap/.venv/lib/python3.13/site-packages/boa/interpret.py", line 252, in loads_partial
    return _loads_partial_vvm(source_code, version, filename)
  File "/my/path/scrap/.venv/lib/python3.13/site-packages/boa/interpret.py", line 286, in _loads_partial_vvm
    return _disk_cache.caching_lookup(cache_key, _compile)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "/my/path/scrap/.venv/lib/python3.13/site-packages/boa/util/disk_cache.py", line 70, in caching_lookup
    res = func()
  File "/my/path/scrap/.venv/lib/python3.13/site-packages/boa/interpret.py", line 277, in _compile
    return VVMDeployer.from_compiler_output(compiler_output, filename=filename)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/my/path/scrap/.venv/lib/python3.13/site-packages/boa/contracts/vvm/vvm_contract.py", line 41, in from_compiler_output
    abi = compiler_output["abi"]
          ~~~~~~~~~~~~~~~^^^^^^^
TypeError: string indices must be integers, not 'str'
```

`favorites.vy` is a very minimal contract.

### PR DESCRIPTION
**Pull request by trocher**
### What I did

- Bump vvm version.
- Fix vvm version detection. The regex was too restrictive and would not match for example `# @version ^0.3.7` or  `# pragma version >=0.3.10`.
- Fix #342 

### How I did it

Uses `vvm` version's detector which uses `SpecifierSet`.

### How to verify it

See the updated test that would fail without the new changes.

### Description for the changelog

    fix vvm version detection

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: boa/interpret.py [hunk 1]
```diff
@@ -34,6 +35,7 @@
 from boa.rpc import json
 from boa.util.abi import Address
 from boa.util.disk_cache import DiskCache
+from boa.util.version import detect_version
```
Comment (charles-cooper) [inline boa/interpret.py:38]: should we actually use the one from vvm?

Comment (DanielSchiavini) [reply boa/interpret.py:38]: Charles is talking about this: https://github.com/vyperlang/vvm/pull/23

**Comment by trocher** [comment]
- https://github.com/vyperlang/vvm/pull/30 is a blocker for that PR since: 
  - it exposes `detect_version_specifier_set` such that titanoboa can check if the installed vyper version is in the set before calling `detect_vyper_version_from_source`.
  - it relax vvm version check to allow for versions like `>=0.3.10, <0.4.0` or `^0.4.0`
- https://github.com/vyperlang/vvm/pull/29 is not a blocker but is recommanded as vvm compilation will currently fail for versions `>0.3.10`.

**Comment by charles-cooper** [comment]
> * [fix: relax version detection checks vvm#30](https://github.com/vyperlang/vvm/pull/30) is a blocker for that PR since:
>   
>   * it exposes `detect_version_specifier_set` such that titanoboa can check if the installed vyper version is in the set before calling `detect_vyper_version_from_source`.
>   * it relax vvm version check to allow for versions like `>=0.3.10, <0.4.0` or `^0.4.0`
> * [fix `combined_json` for versions >0.3.10 vvm#29](https://github.com/vyperlang/vvm/pull/29) is not a blocker but is recommanded as vvm compilation will currently fail for versions `>0.3.10`.

these are merged and part of vvm 0.3.2 release (https://pypi.org/project/vvm/0.3.2/), so i think we can update this PR now

File: boa/interpret.py [hunk 2]
```diff
@@ -249,8 +254,10 @@ def loads_partial(
     if dedent:
         source_code = textwrap.dedent(source_code)
 
-    version = _detect_version(source_code)
-    if version is not None and version != vyper.__version__:
+    specifier_set = detect_version_specifier_set(source_code)
+    # Use VVM only if the installed version is not in the specifier set
+    if specifier_set is not None and not specifier_set.contains(vyper.__version__):
+        version = detect_vyper_version_from_source(source_code)
```
Comment (charles-cooper) [inline boa/interpret.py:260]: this results in two searches of the source code. i think we can use vvm's _pick_vyper_version instead?

**Review by charles-cooper — APPROVED** [review]
nice work. thanks!

**Pull request merged**